### PR TITLE
INTMDB-781: [Terraform] Parameter Add: retainBackups in Cluster and Advanced_Cluster

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Description

This PR adds support for `retain_backups_enabled` for the advanced cluster and legacy cluster resources.

**Note:** `retain_backups_enabled` was not added to the data sources as it is used only during the deletion of the cluster resource. 

Ticket: [INTMDB-781](https://jira.mongodb.org/browse/INTMDB-781)

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [x] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code

### Acceptance Testing
**TestAccClusterRSCluster_basicAWS_simple**
```
Running tool: /Users/andrea.angiolillo/.asdf/shims/go test -timeout 300000s -run ^TestAccClusterRSCluster_basicAWS_simple$ github.com/mongodb/terraform-provider-mongodbatlas/mongodbatlas

ok  	github.com/mongodb/terraform-provider-mongodbatlas/mongodbatlas	1609.446s

```

**TestAccClusterRSAdvancedCluster_singleProvider**
```
2023-06-06T17:37:40.967+0100 [DEBUG] sdk.helper_resource: Finished TestCase: test_name=TestAccClusterRSAdvancedCluster_singleProvider
PASS
```

### I run the example
The snapshot was not deleted after the cluster was removed.

Screenshot 2023-06-06 at 18 46 09

### CODE REVIEW

**Review by martinstibbe — APPROVED** [review]
LGTM

File: examples/atlas-backup-snapshot-restore-job/point-in-time/versions.tf [hunk 1]
```diff
@@ -1,9 +1,8 @@
 terraform {
   required_providers {
     mongodbatlas = {
-      source  = "mongodb/mongodbatlas"
-      version = "~> 1.4.6"
```
Comment (Zuhairahmed) [inline examples/atlas-backup-snapshot-restore-job/point-in-time/versions.tf:5]: this is great fix @andreaangiolillo, in future for all examples we should just default to pulling latest version of Terraform Atlas Provider

**Review by Zuhairahmed — APPROVED** [review]
LGTM!

**Review by colm-quinn — APPROVED** [review]
LGTM

File: examples/atlas-backup-snapshot-restore-job/point-in-time-advance-cluster/README.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+Variables Required:
+- `org_id`: ID of atlas organization
+- `project_name`: Name of the project
+- `cluster_name`: Name of the cluster
+- `point_in_time_utc_seconds`: Point in time to restore to, a number of seconds since unix epoch.
+- `retain_backups_enabled`: Flag that indicates whether to retain backup snapshots for the deleted dedicated cluster.
+
+In order to utilize the backup restore job via point in time, fist you need a backup with which to restore.
+This example has been configured to only create the backup restore if `point_in_time_utc_seconds` is a non-zero number.
+As such, utilize the following example `terraform.tfvars` and pseudo-code to execute a workign example:
```
Comment (zach-carr) [inline examples/atlas-backup-snapshot-restore-job/point-in-time-advance-cluster/README.md]: [typo]
```suggestion
As such, utilize the following example `terraform.tfvars` and pseudo-code to execute a working example:
```

File: examples/atlas-backup-snapshot-restore-job/point-in-time-advance-cluster/main.tf [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+      provider_name = "AWS"
+      region_name   = "US_EAST_1"
+      priority      = 7
+    }
+  }
+
+  backup_enabled         = true                       # enable cloud backup snapshots
+  pit_enabled            = true                       # Flag that indicates whether the cluster uses continuous cloud backups
+  retain_backups_enabled = var.retain_backups_enabled # keep the backup snapshopts once the cluster is deleted
```
Comment (zach-carr) [inline examples/atlas-backup-snapshot-restore-job/point-in-time-advance-cluster/main.tf]: [typo]
```suggestion
  retain_backups_enabled = var.retain_backups_enabled # keep the backup snapshots once the cluster is deleted
```

**Review by zach-carr — APPROVED** [review]
LGTM, just found a couple typos

**Pull request merged**